internal/uploader/auth_utils.go

```diff
@@ -0,0 +1,96 @@
+package uploader
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	openshiftTokenKey = "cloud.openshift.com"
+)
+
+type serializedAuthMap struct {
+	Auths map[string]serializedAuth `json:"auths"`
+}
+type serializedAuth struct {
+	Auth  string `json:"auth"`
+	Email string `json:"email,omitempty"`
+}
+
+func getPullSecret(db *gorm.DB, clusterID *strfmt.UUID, k8sclient k8sclient.K8SClient, key string) (*serializedAuth, error) {
+	data, exists := getOCMPullSecret(k8sclient)
+	if !exists {
+		data = getTokenFromCluster(db, clusterID)
```

Comment:
I thought the failure point could exist in [line 38](https://github.com/openshift/assisted-service/pull/4880/files/e42e509a56ae73ac17cda10a6a3412d8bdb9df6b#diff-be3c5c34d4670110a2ffa6cd391d344a1c14941e7813cce8eb05cd9e92faf686R38) where if `data` is `nil` after both of these calls, it would error out here. 

Do you think it's better to exit early before even trying to unmarshal?

Reply:
I don't think it is needed to be logged here as well if it is already handled by the caller.

Reply:
Gotcha, thanks for confirming!